scrapy/utils/url.py

```diff
@@ -103,3 +103,16 @@ def guess_scheme(url):
         return any_to_uri(url)
     else:
         return add_http_if_no_scheme(url)
+
+
+def trim_href_attribute(href):
```

Comment:
As for code readability, I think the part of link extractors where it's used is already self-explanatory, that it's stripping the attribute value. (And it could be used by some users for non-href, although I don't know how that's defined in the specs, for `src` for example)